### PR TITLE
Fix: Connection Type

### PR DESCRIPTION
**Pull request by doctrino**
## Description
Currently breaks the SDK

## Checklist:
- [x] Tests added/updated.
- [ ] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [x] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [x] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

**Review by erlendvollset — APPROVED** [review]
LGTM. Although I'd like to see a more systemic approach to solving this. It's a reoccurring issue in the DM data classes.

File: cognite/client/data_classes/data_modeling/views.py [hunk 1]
```diff
@@ -386,17 +387,19 @@ class SingleHopConnectionDefinition(ConnectionDefinition):
...
-            output.edge_source = ViewId.load(edge_source)
-        return output
+        return cls(
+            type=DirectRelationReference.load(data["type"]),
+            source=ViewId.load(data["source"]),
+            name=data.get("name"),
+            description=data.get("description"),
+            edge_source=(edge_source := data.get("edgeSource")) and ViewId.load(edge_source),
+            direction=data.get("direction"),  # type: ignore[arg-type]
+            connection_type=to_snake_case(data.get("connectionType")),  # type: ignore[arg-type]
```
Comment (erlendvollset) [inline cognite/client/data_classes/data_modeling/views.py:401]: These will never be null, so we shouldn't use `.get`
```suggestion
            direction=data["direction"]
            connection_type=to_snake_case(data["connectionType"])
```

File: cognite/client/data_classes/data_modeling/views.py [hunk 2]
```diff
@@ -439,17 +442,19 @@ class SingleHopConnectionDefinitionApply(ConnectionDefinitionApply):
...
-        return output
+    def load(cls, data: dict[str, Any]) -> SingleHopConnectionDefinitionApply:
+        return cls(
+            type=DirectRelationReference.load(data["type"]),
+            source=ViewId.load(data["source"]),
+            name=data.get("name"),
+            description=data.get("description"),
+            edge_source=(edge_source := data.get("edgeSource")) and ViewId.load(edge_source),
+            direction=data.get("direction"),  # type: ignore[arg-type]
+            connection_type=to_snake_case(data.get("connectionType")),  # type: ignore[arg-type]
```
Comment (erlendvollset) [inline cognite/client/data_classes/data_modeling/views.py:456]: Same here

File: cognite/client/data_classes/data_modeling/views.py [hunk 3]
```diff
@@ -386,17 +387,19 @@ class SingleHopConnectionDefinition(ConnectionDefinition):
     description: str | None = None
     edge_source: ViewId | None = None
     direction: Literal["outwards", "inwards"] = "outwards"
+    connection_type: Literal["multi_edge_connection"] = "multi_edge_connection"
```
Comment (erlendvollset) [inline cognite/client/data_classes/data_modeling/views.py:390]: Hmm. Is it really necessary to snake case the enum?

Comment (doctrino) [reply cognite/client/data_classes/data_modeling/views.py:391]: Is just that we do it as far as I know everywhere else in the SDK.

**Review by erlendvollset — APPROVED** [review]

**Pull request merged**